moveit_ros/visualization/CMakeLists.txt

```diff
@@ -65,11 +65,11 @@ catkin_package(
     planning_scene_rviz_plugin/include
     motion_planning_rviz_plugin/include
     trajectory_rviz_plugin/include
-    ${EIGEN3_INCLUDE_DIRS}
   CATKIN_DEPENDS
+    EIGEN3
```

Comment:
this is wrong. `EIGEN3` has to go into a `DEPENDS` section
Same for most of the other places you changed.